src/python/pants/option/global_options.py

```diff
@@ -1611,3 +1630,25 @@ class ProcessCleanupOption:
     """
 
     val: bool
+
+
+def maybe_warn_python_macros_deprecation(bootstrap_options: OptionValueContainer) -> None:
+    if bootstrap_options.is_default("use_deprecated_python_macros"):
+        warn_or_error(
+            "2.11.0.dev0",
+            "the option `--use-deprecated-python-macros` defaulting to true",
```

Comment:
One thing that would be good to fix before `2.10.0` is that this warning currently triggers in repos without the python backend enabled.